pkg/analyze/host_analyzer_test.go

```diff
@@ -10,25 +11,29 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func collectorToBytes(collector any) []byte {
```

Comment:
nit: You could pass in `t` and fail if err like so

```go
func collectorToBytes(t *testing.T, collector any) []byte {
	jsonData, err := json.Marshal(collector)
	require.NoError(t, err) 
	return jsonData
}
```

then call the function like so

```
Data: collectorToBytes(t, collect.HostOSInfo{...
```

Reply:
Sick, thanks